csharp/ql/test/query-tests/Telemetry/DatabaseQuality/Quality.cs

```diff
@@ -16,13 +16,43 @@ public Test()
 
         new Test()
         {
-            MyProperty4 = { 1, 2, 3 }
+            MyProperty4 = { 1, 2, 3 },
+            MyProperty6 = { [1] = "" }
         };
+
+        Event1.Invoke(this, 5);
+
+        var str = "abcd";
+        var sub = str[..3];
```

Comment:
Maybe it is worth adding a comment about "false positive", unless you intend to look into improving the range/slice indexing? (similar comment for the other indexer cases below).

Reply:
I consider all the findings in https://github.com/github/codeql/pull/18166/files#diff-18cb5a31dd2ca102defe177591890884bf5bc3a6d3e7490823990682b7e24af4 to be cases that we'd need to cover. 

Let me add the comments in the test file to make this more straightforward.